src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -75,6 +75,8 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, PageRouteMode
             {
                 var descriptor = new PageActionDescriptor
                 {
+                    ActionConstraints = selector.ActionConstraints.ToList(),
```

Comment:
Why `ToList()`? In particular, why use `ToList()` in two of the the three additional copies?